google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -1266,11 +1271,7 @@ public Timestamp write(Iterable<Mutation> mutations) throws SpannerException {
     @Override
     public CommitResponse writeWithOptions(
         Iterable<Mutation> mutations, TransactionOption... options) throws SpannerException {
-      try {
-        return get().writeWithOptions(mutations, options);
-      } finally {
-        close();
-      }
+      return get().writeWithOptions(mutations, options);
```

Comment:
This method is no longer responsible for closing the session, instead that is done by the `runWithSessionRetry` method.